components/IndoorDirections/TypesOfDirections/DifferentBuildingDirections.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import { Line, G } from 'react-native-svg';
+import { ClassGraph } from '../../../constants/ClassGraph';
+import { dijkstra, getFloorNumber, changeClassName } from '../Dijkstra/DijkstraAlgorithm';
+
+export function DifferentBuildingDirections(props) {
+    const rooms = props.rooms;
+    const graph = ClassGraph();
+    var goTo = props.from;
+    var tempGoTo = props.from;
+
+    if (props.from.includes(" ")) {
+        tempGoTo = goTo;
+        goTo = changeClassName(props.to);
+    }
+
+    if (props.to.includes(" ")) {
+        tempGoTo = goTo;
+        goTo = changeClassName(props.from);
+    }
+
+    const pathToElevator = dijkstra(graph, goTo, "elevator").path;
+    const pathToClass = dijkstra(graph, "elevator", "exit").path;
+
+    var getNextRoomElevator = (index) => {
+        if (index < pathToElevator.length){
+            return pathToElevator[++index];
+        }
+    }
+
+    var getNextRoomClass = (index) => {
+        if (index < pathToClass.length){
+            return pathToClass[++index];
+        }
+    }
+
+    const linesToElevator = [];
+    pathToElevator.forEach((element, index) => {
+        if (index < pathToElevator.length - 1) {
+            linesToElevator.push({x1: rooms[element].nearestPoint.x, y1: rooms[element].nearestPoint.y, x2: rooms[getNextRoomElevator(index)].nearestPoint.x, y2: rooms[getNextRoomElevator(index)].nearestPoint.y})
+        }
+    });
```

Comment:
Is this responsible for drawing the lines to the elevator?

Reply:
When you compare length of the constant pathToElevator to the index, how does length of pathToElevator gets its value?  Same thing for the PathToClass